tests/unit/test_discovery_client.py

```diff
@@ -236,19 +240,37 @@ def test___get_discovery_service_address___key_file_not_exist___throws_timeouter
         "ni_measurementlink_service._internal.discovery_client._START_SERVICE_TIMEOUT", 5.0
     )
     mocker.patch(
-        "ni_measurementlink_service._internal.discovery_client._open_key_file", side_effect=IOError
+        "ni_measurementlink_service._internal.discovery_client._open_key_file", side_effect=OSError
     )
     mocker.patch(
         "ni_measurementlink_service._internal.discovery_client._get_registration_json_file_path",
         return_value=temp_registration_json_file_path,
     )
     mocker.patch("subprocess.Popen")
 
-    with pytest.raises(IOError) as exc_info:
+    with pytest.raises(OSError) as exc_info:
         _get_discovery_service_address()
     assert exc_info.type is TimeoutError
 
 
+@pytest.mark.parametrize(
+    "windows_error_code", [2, 3]
+)  # ERROR_FILE_NOT_FOUND = 2, ERROR_PATH_NOT_FOUND = 3
+def test___key_file_not_exist___open_key_file___raises_file_not_found_error(
+    mocker: MockerFixture, temp_discovery_key_file_path: pathlib.Path, windows_error_code: int
+) -> None:
+    if sys.platform != "win32":
+        pytest.skip("Windows-only test")
+    else:
+        mocker.patch(
+            "win32file.CreateFile",
+            side_effect=win32file.error(windows_error_code, None, None),
```

Comment:
This is causing mypy errors in my dev branches.

https://github.com/ni/measurementlink-python/pull/422

Reply:
I'm updating the PR workflow to check Windows-specific code.

https://github.com/ni/measurementlink-python/pull/423

Reply:
Sure.